src/rust/ide/lib/view/src/graph_editor/component/documentation_view.rs

```diff
@@ -0,0 +1,159 @@
+//! Documentation view visualization generating and presenting Enso Documentation under
+//! the documented node.
+
+use crate::prelude::*;
+
+use crate::graph_editor::component::visualization::*;
+use crate::graph_editor::component::visualization;
+
+use enso_frp as frp;
+use ensogl::display::DomSymbol;
+use ensogl::display::scene::Scene;
+use ensogl::display;
+use ensogl::system::web;
+use ensogl::system::web::StyleSetter;
+
+/// Generates Documentation View stylesheet.
+pub fn get_doc_style() -> String {
+    format!("<style>{}</style>", include_str!("documentation_view/style.css"))
+}
+
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct DocumentationViewModel {
+    logger : Logger,
+    dom    : DomSymbol,
+    size   : Rc<Cell<Vector2>>,
+}
+
+impl DocumentationViewModel {
+    /// Constructor.
+    fn new(scene:&Scene) -> Self {
+        let logger = Logger::new("DocumentationView");
+        let div    = web::create_div();
+        let dom    = DomSymbol::new(&div);
+        let screen = scene.camera().screen();
+        let size   = Rc::new(Cell::new(Vector2(290.0,screen.height - 30.0)));
+
+        dom.dom().set_style_or_warn("white-space"     ,"normal"                             ,&logger);
+        dom.dom().set_style_or_warn("overflow-y"      ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("overflow-x"      ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("background-color","rgba(255, 255, 255, 0.85)"          ,&logger);
+        dom.dom().set_style_or_warn("padding"         ,"5px"                                ,&logger);
+        dom.dom().set_style_or_warn("pointer-events"  ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("border-radius"   ,"14px"                               ,&logger);
+        dom.dom().set_style_or_warn("width"           ,format!("{}px", 290)                 ,&logger);
+        dom.dom().set_style_or_warn("height"          ,format!("{}px", screen.height - 30.0),&logger);
+
+        scene.dom.layers.main.manage(&dom);
+        DocumentationViewModel{dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    const PLACEHOLDER_STR: &'static str = "<h3>Enso Documentation Viewer</h3>\
+                                           <p>No documentation available</p>";
```

Comment:
such things need documentation. In particular, when this caption will be visible?

Reply:
It is the content in the documentation view when no data has been yet received - think of it like in Xcode when nothing is opened, it says "No editor", or in right panel - "No selection"